File: C-Sharp (MonoGame)/C-Sharp (MonoGame)/Assets/Enemy.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_Sharp__MonoGame_
{
    class Enemy : Sprite
    {
        int enemyCentre;
        public int score = 0;

        public Enemy(Texture2D newTexture)
        {
            texture = newTexture;
            position = new Vector2(screenWidth - texture.Width, (screenHeight - texture.Height) / 2);
        }

        public void Update(int ballCentre)
        {
            rectangle = new Rectangle((int)position.X, (int)position.Y, texture.Width, texture.Height);

            if (ballCentre > enemyCentre)
            {
                position.Y += 5;
            }
            if (ballCentre < enemyCentre)
            {
                position.Y -= 5;
            }
            if (position.Y <= 0)
                position.Y = 0;
            if (position.Y + rectangle.Height >= screenHeight)
                position.Y = screenHeight - rectangle.Height;

            enemyCentre = (int)position.Y + (rectangle.Height / 2);
        }
    }
}

File: С-PlusPlus/С-PlusPlus/Enemy.cpp
#include "Main.h"

Enemy::Enemy(int eWidth, int eHeight)
{
	mPosY = (screenHeight - eHeight) / 2;
	mCollider.x = screenWidth - eWidth;
	mCollider.w = eWidth;
	mCollider.h = eHeight;
}

void Enemy::render()
{
	sEnemy.render(mCollider.x,mPosY);
}

void Enemy::move(int ballCentre)
{
	enemyCentre = mPosY + (mCollider.h / 2);
	if (ballCentre > enemyCentre) mPosY += 5;
	if (ballCentre < enemyCentre) mPosY -= 5;
	if (mPosY <= 0) mPosY = 0;
	if (mPosY + mCollider.h >= screenHeight) mPosY = screenHeight - mCollider.h;
	mCollider.y = mPosY;
}